include/kokkos_kernels/impl/compute_stiffness_interaction.hpp

```diff
@@ -8,7 +8,24 @@
 namespace specfem {
 namespace kokkos_kernels {
 namespace impl {
-
+/**
+ * @brief Compute the source interaction for the given medium.
+ *
+ * This function computes the source interaction for the specified medium type
+ * and properties. It is specialized for different dimension tags, medium tags,
+ * property tags, and boundary tags.
+ *
+ * @tparam DimensionTag Spatial dimension (2D/3D)
+ * @tparam WavefieldType Simulation wavefield type (e.g., forward, adjoint,
+ * backward)
+ * @tparam NGLL Number of GLL points
+ * @tparam MediumTag Medium type (e.g., elastic, acoustic)
+ * @tparam PropertyTag Material property type (e.g., isotropic, anisotropic)
+ * @tparam BoundaryTag Boundary condition type (e.g., free_surface, absorbing)
+ *
+ * @param assembly SPECFEM++ assembly object.
+ * @param timestep Time step for which the source interaction is computed
```

Comment:
```suggestion
 * @param istep Time step for which the source interaction is computed
```

Note that I actually prefer the variable name ``timestep``